Plans section: Modernized header and navbar / part 1

chriskmnds opened the pull request:
### Proposed Changes

Partially addresses https://github.com/Automattic/martech/issues/1463

This brings the "modernized" header & navbar by default to `/plans/my-plan` & `/plan` (not behind a feature flag). This is another temp solution with follow-ups in TODO list below.

_Notes:_ Plans header & navigation are used for [jetpack-plans](https://github.com/Automattic/wp-calypso/tree/trunk/client/my-sites/plans/jetpack-plans) code, but the changes should not affect these (something for a follow-up)

#### Media

with some of the states I notice for the plans pages (info & upsell notices):

_Old `/plans`_

<img width="700" alt="Screenshot 2023-02-06 at 4 25 02 PM" src="https://user-images.githubusercontent.com/1705499/216997389-4ab1c688-857d-4958-81a4-cbd73af1b934.png">

_New `/plans`_

<img width="700" alt="Screenshot 2023-02-06 at 4 23 42 PM" src="https://user-images.githubusercontent.com/1705499/216997457-c062689e-2417-4522-9ced-c2bfcef61345.png">

_New `/plans` free site_

<img width="700" alt="Screenshot 2023-02-06 at 5 12 03 PM" src="https://user-images.githubusercontent.com/1705499/217009326-c5995696-26ae-4186-a241-3fad2e70fc5f.png">


_`/plans/my-plan`_

<img width="700" alt="Screenshot 2023-02-06 at 4 24 30 PM" src="https://user-images.githubusercontent.com/1705499/216997597-9fc7b1b1-f348-4e16-94e5-b8010dc39895.png">

### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Navigate to `/plans/[site_slug]?flags=onboarding/2023-pricing-grid`
- Confirm the changes
- Do the same without the flag `/plans/[site_slug]?flags=-onboarding/2023-pricing-grid`
- Confirm no changes in `jetpack-plans` (and please post instructions in a comment on how to test that while at it 😄)

### TODO

- [x] Extract into a layout component to allow reuse | [issue](https://github.com/Automattic/wp-calypso/issues/73102)
- [x] Extract navigation into a component to allow reuse | [issue](https://github.com/Automattic/wp-calypso/issues/73102)
- [x] Consider navbar inside header to reduce some of the complexity | [issue](https://github.com/Automattic/wp-calypso/issues/73102)

### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to Automattic/martech#1463

chriskmnds committed:
Plans section: Modernized header and navbar / part 1

chriskmnds committed:
cleanup

chriskmnds committed:
renames, memo

chriskmnds committed:
nav cleanup

chriskmnds committed:
header cleanup

chriskmnds committed:
nav cleanup

chriskmnds committed:
cleanup, drop header shadow on free

chriskmnds commented:
@lucasmendes-design for design review :-)

jdc91 commented:
Ah this looks elegant 💯 @chriskmnds

lucasmendes-design commented:
Hi there, 

**Monthly:**

When I selected monthly, the copy is "Save 56% by paying annually", but I believe It should be "per month".

<br class="Apple-interchange-newline">
<img width="1136" alt="image" src="https://user-images.githubusercontent.com/98343583/217118381-7a000d44-6d57-429d-bfd7-c6c44c102fd9.png">

**Visitor stats:**

Sometimes, this feature is misaligned. I checked again and it's fine 🤔 

<img width="434" alt="image" src="https://user-images.githubusercontent.com/98343583/217118712-fafee29e-3256-4eb6-9052-58b37bb15502.png">
<img width="549" alt="image" src="https://user-images.githubusercontent.com/98343583/217120837-db46c800-f320-45a1-bbf8-d853e15daa13.png">



**Sticky header:** 

Are we going to have a sticky header in Calypso? 

**Check** 

I checked on the design and the y'all did the right thing but looking at the plans comparison grid right now. I think the check ✔ mark is too big. Is it simple to change from 24px to 20px? We can also upload that on different pages. 

<img width="301" alt="image" src="https://user-images.githubusercontent.com/98343583/217119132-f94c0632-dbfd-4890-b34b-d5da2394f13e.png">


**Chevron icon**

jdc91 commented:
@lucasmendes-design The generic feedback you have raised is important but needs to be raised elsewhere. This PR is focussed on the header area as shown below. Since there is not much feedback on that area, I hope we can assume that the design is pixel perfect there?
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/3422709/217147292-2dd1c0e7-6ac9-4494-8fd5-18d3edb75db6.png">

I will try to carry this feedback to a ticket, but, going forward, please do raise generic feedback on the task board  :) https://github.com/orgs/Automattic/projects/552/views/1

southp commented on client/my-sites/plans/current-plan/index.jsx:
```diff
@@ -232,71 +233,75 @@ class CurrentPlan extends Component {
-
-					<TrackComponentView eventName="calypso_plans_my_plan_view" />
+				{ selectedSiteId && (
+					// key={ selectedSiteId } ensures data is refetched for changing selectedSiteId
```
I know it's been this way for ages, but let's remove this since it doesn't make sense anymore.

southp commented on client/my-sites/plans/main.jsx:
```diff
@@ -184,38 +187,39 @@ class Plans extends Component {
+								wideLayout={ ! is2023OnboardingPricingGrid }
+							>
+								{ domainAndPlanPackage && <DomainAndPlanUpsellNotice /> }
+								{ ! canAccessPlans && (
```
This section is nested under `canAccessPlans && (...`, which means it will never be reached. From the original code, we should move this section outside of this `canAccessPlans` expression.

chriskmnds replied:
Good catch :-)

southp commented on client/my-sites/plans/navigation.scss:
```diff
@@ -0,0 +1,89 @@
+		box-shadow: inset 0 -1px 0 #0000000d;
+		background-color: var(--studio-white);
+
+		// .section-nav__panel {
```
If there is no good reason for keeping it, let's remove this commented code block :)

southp reviewed (commented):
Outside of the inline comments, here are a few more things I'd like to raise:

Comparing My Plans and Plans tab, the former has two horizontal lines while the latter has only one. Which one is the intended result?
<img width="988" alt="截圖 2023-02-07 上午11 05 29" src="https://user-images.githubusercontent.com/1842898/217161649-2e14d65f-98bf-49fd-9773-5a697c063e9f.png">
<img width="981" alt="截圖 2023-02-07 上午11 05 53" src="https://user-images.githubusercontent.com/1842898/217161661-c18eabcc-24ce-4f54-bf27-075d22d4ec1f.png">

Testing the Jetpack plans page, the Plans tab is unaffected while My Plan tab is affected, presenting inconsitent look upon switching tabs:

https://user-images.githubusercontent.com/1842898/217161882-0f36268a-6454-470e-9219-1cf64fc35adb.mov

Finally, the idea of `ModernizedLayout` component is quite intersting and works in general. However, I'm a bit confused that we have two ways of adding the "modernized" classname: either adding it directly to the component like `PlansHeaders` does or conditionally adding it through a prop like `PlansNavigation` does. I'm wondering if we can make it more consistent? This is more of a question than a blocking comment, so please feel free to keep it the way it is :)

southp commented:
Ohhhh right, I forgot about this:

> Confirm no changes in jetpack-plans (and please post instructions in a comment on how to test that while at it 😄)

Here goes:

1. Go to https://jurassic.ninja to create a Jetpack site.
2. Once it's done, go to its wp-admin and access the Jetpack dashboard there. 
3. There, you should be able to find a button to connect the site to your WordPress.com account. Click it and follow the instructions.
4. Go back to calypso, and click "switch site" to change the selected site to this Jetpack site.
5. Now the plans tab will present you with `jetpack-plans` component :)

chriskmnds committed:
no jetpack. header shadow fix

chriskmnds committed:
cleanup. consolidate nav/layout style overrides

chriskmnds commented on client/my-sites/plans/modernized-layout.tsx:
```diff
@@ -0,0 +1,150 @@
+	const verticalMargin = '32px';
+
+	const globalOverrides = css`
+		.${ subSection ?? section }__section-header {
```
These variable class names are (ought to be) really temp stuff. Ideally, what's mapped on the front e.g. `.plans__section-header` should be searchable in the code base.

chriskmnds replied:
refactored back to static classes for now

chriskmnds committed:
cleanup. modernizer class names

chriskmnds commented:
@southp @jdc91 Thanks for the thorough review, and the detailed JP instructions. I believe I've addressed your concerns, but pls do take another look. I'll do few more tests on my end.

> Comparing My Plans and Plans tab, the former has two horizontal lines while the latter has only one. Which one is the intended result?

There should only be one. Just clumsy me :p

> Finally, the idea of ModernizedLayout component is quite intersting and works in general. However, I'm a bit confused that we have two ways of adding the "modernized" classname: either adding it directly to the component like PlansHeaders does or conditionally adding it through a prop like PlansNavigation does. I'm wondering if we can make it more consistent? This is more of a question than a blocking comment, so please feel free to keep it the way it is :)

Yeah, it's a bit of a soup. I think I wanted to keep them closer to how they were defined elsewhere (one being part of the nav component styles, the other being actual section overrides), but that idea was short-lived. I've consolidated the styles into `ModernizedLayout` now and removed the smart classes along. It's all still temporary. Ideally, we'd want to follow up and extract these overrides/styles to a place and form that are convenient for reuse in other sections (and update all the sections to sync! 😬)

chriskmnds commented on client/my-sites/plans/main.jsx:
```diff
@@ -186,36 +189,36 @@ class Plans extends Component {
 		return (
-			<div className={ is2023OnboardingPricingGrid ? 'is-2023-pricing-grid' : '' }>
+			<div>
+				<ModernizedLayout dropShadowOnHeader={ isFreePlan( currentPlanSlug ) } />
```
should probably check for Jetpack here also, for sanity

chriskmnds committed:
jetpack check in plans. for sanity

ddc22 commented on client/my-sites/plans/current-plan/index.jsx:
```diff
@@ -329,5 +334,6 @@ export default connect( ( state, { requestThankYou } ) => {
 		showThankYou: requestThankYou && isJetpackNotAtomic,
 		scheduleId: getConciergeScheduleId( state ),
 		eligibleForProPlan,
+		isJetpack,
```
This will also flag atomic sites as Jetpack sites and the Modernized header will not be visible. 

![image](https://user-images.githubusercontent.com/3422709/217307294-5c8978fd-d533-401e-a696-13c1f9793099.png)


I think the flag on line 318 accounts for this and can be used for our use case as well.

https://github.com/Automattic/wp-calypso/blob/fd4d73e86ebc9dc8af48c2fd0134406ae7b4094c/client/my-sites/plans/current-plan/index.jsx#L318

chriskmnds replied:
Ah! Good catch. totally ignored this.

chriskmnds replied:
(and funnily enough, I was testing with an atomic site to confirm the jetpack condition at some point. so I'm the worst! 😆 )

ddc22 reviewed (changes requested):
Overall looks good but a minor fix is required for Atomic sites.

lucasmendes-design commented:
Hi there, there is one detail that I believe we can fix it. 

On this page, the common alignment is center-align. To give an example here is how stats and themes page look like with the new header: 

**Stats page:**
<img width="1001" alt="image" src="https://user-images.githubusercontent.com/98343583/217334914-f2fbb707-39ac-41ec-823e-f87fecf9de4a.png">

Themes page:
<img width="730" alt="image" src="https://user-images.githubusercontent.com/98343583/217336183-9ac55d8c-5b50-4a93-b812-41634030728c.png">

I would change the size of the notification to follow the size of the plans grid and the header itself. [Figma file](https://www.figma.com/file/xkhhUlDowF13dRoXJqlRDM?node-id=1:95#363744244)

**1024px**
<img width="698" alt="image" src="https://user-images.githubusercontent.com/98343583/217337162-e3debbd9-76e6-4e03-b742-ebe6f584e11e.png">

**1440px**
<img width="780" alt="image" src="https://user-images.githubusercontent.com/98343583/217337211-9475773a-59cb-4fbb-a4b7-d1f51c34d7bf.png">

**Notification:**
<img width="979" alt="image" src="https://user-images.githubusercontent.com/98343583/217337389-44d0fe4d-31d6-4316-ac33-dc6369c52daf.png">

chriskmnds commented:
> I would change the size of the notification to follow the size of the plans grid and the header itself.

@lucasmendes-design I'm not exactly sure how that's _not_ the case right now. Notification seems to be aligned with the same padding, so it follows the size of the header? 🤔 

<img width="1323" alt="Screenshot 2023-02-08 at 11 12 01 AM" src="https://user-images.githubusercontent.com/1705499/217486799-a67ed7ee-3d2a-45ac-8f98-e9d4763b8c70.png">

chriskmnds committed:
update atomic/jp checks

chriskmnds committed:
cleanup/fix nav styles

chriskmnds committed:
cleanup styles. first-child -> first-of-type

chriskmnds commented on client/my-sites/plans/main.jsx:
```diff
@@ -241,5 +248,6 @@ export default connect( ( state ) => {
 		showTreatmentPlansReorderTest: isTreatmentPlansReorderTest( state ),
 		plansLoaded: Boolean( getPlanSlug( state, getPlan( PLAN_FREE )?.getProductId() || 0 ) ),
 		is2023OnboardingPricingGrid,
+		isJetpackNotAtomic: isJetpackSite( state, selectedSiteId, { treatAtomicAsJetpackSite: false } ),
```
Apparently, there's a `options.treatAtomicAsJetpackSite` parameter that can be set on this selector (and avoid having to [combine two selectors](https://github.com/Automattic/wp-calypso/pull/72960/files#diff-9a7bb16689c221cf314662dc7929cea29ae46431f9bc11fd4123e233b0b208d2R317-R318) for it as in the other file). Not sure I'm missing something though. @jdc91

ddc22 replied:
Nice this is a better approach!

chriskmnds commented:
@jdc91 Thanks for reviewing. Sorry that I'm dragging you into spotting errors for me 😬 

Let's give this another review pls, I also noticed console errors with the CSS parts that needed to resolve. will do a few more tests on my end

ddc22 reviewed (approved):
LGTM :shipit:

lucasmendes-design commented:
Hi there, Dustin Reagon mention [that here (Slack). ](https://a8c.slack.com/archives/C02FMH4G8/p1675875224401949)

The problem with this layout is that the header with the copy "Plans" and "See and compare features available on each WordPress.com plan" is misaligned. 
![image](https://user-images.githubusercontent.com/98343583/217629391-93d9e1db-3d7f-4c9c-860b-104902c7db95.png)

Everything should be center-aligned, not only the plans grid. Domains example below:
![image](https://user-images.githubusercontent.com/98343583/217629657-cc384444-08ae-4efc-a304-8c3bf8c4d226.png)

cc: @jdc91
